### PR TITLE
Add new blog post for website search functionality

### PR DESCRIPTION
**Pull request by zelinh**
### Description
Add new blog post for website search functionality
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/3379

### Check List
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

**Comment by pajuric** [comment]
@zelinh - Thanks for the contribution. I'll reach out to you on Slack.

**Comment by pajuric** [comment]
@zelinh - Please use sentence case for headlines/titles and section headers. For example, "Enhanced Search Infrastructure" should be "Enhanced search infrastructure."

**Comment by zelinh** [comment]
> @zelinh - Please use sentence case for headlines/titles and section headers. For example, "Enhanced Search Infrastructure" should be "Enhanced search infrastructure."

Updated in the new commit. Thanks.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+## The new search functionality
+
+The previous search system had limitations related to scalability, maintenance, and user experience. With this update, we're moving to Amazon OpenSearch Service for a more reliable and scalable infrastructure, and we've added new search features to make your experience faster and more efficient.
+
+**Here's what's new:**
+
+- [**Improved infrastructure**](#better-search-infrastructure) to eliminate deployment and indexing issues.
+- **No more duplicate results**, giving you cleaner and more accurate search results.
+- A [**new search results page**](#new-search-results-page) that goes beyond the dropdown format.
+- **Filters for documentation, blogs, and events** to help you focus on relevant content.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:26]:
```suggestion
- **Filters for documentation, blog posts, and events** to help you focus on relevant content.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+- The ability to **search archived versions** of documentation.
+
+## Better search infrastructure
+
+We've restructured the backend of our search system to improve both performance and relevance. Now, content is indexed by documentation version, so your search results are version-specific. This reduces clutter and duplication, making it easier for you to find the right information quickly.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:31]:
```suggestion
We've restructured the backend of our search system to improve both performance and relevance. Now content is indexed by documentation version, so your search results are version specific. This reduces clutter and duplication, making it easier for you to find the right information quickly.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+
+We've also upgraded from basic query strings to `match` and `multi_match` queries, which provide more precise results by searching across multiple fields. Additionally, we've applied `indices_boost` to prioritize documentation in search results, further improving relevance.
+
+## New search results page
+
+We've moved away from the simple dropdown search interface to a full search results page, offering a more powerful and user-friendly way to navigate search results. This page gives you more control over your search and makes it easier to refine your results.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:37]:
```suggestion
We've moved away from the simple dropdown search interface in favor of a full search results page, offering a more powerful and user-friendly way to navigate search results. This page gives you more control over your search experience and makes it easier to refine your results.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+## Key features
+
+Here's a look at the key improvements and new features that enhance your search experience on OpenSearch.org:
+
+- **Dedicated search results page**:
+    After submitting a search, you'll be taken to a new page that shows a more comprehensive list of results than the previous dropdown. You can browse through these results more easily, without leaving the page.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:44]:
```suggestion
    After submitting a search, you'll be taken to a new page that shows a more comprehensive list of results than was provided by the previous dropdown. You can browse through these results more easily, without leaving the page.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+
+- **Persistent search box**:
+    The search box stays at the top of the page, allowing you to refine your query or enter new keywords without losing your current search context.
+
+- **Advanced filters**:
+    The new page includes filtering options to help you focus your search. You can filter by:
+    - **Documentation**: Narrow down your search to technical documentation.
+    - **Blogs**: View relevant blog posts.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:52]: "Blogs" should be "Blog posts", but if this is the actual name of the filter, then fine to leave as is.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+    - **Events**: Search through events and announcements.
+    - **All**: Search across all types of content.
+
+    You can also filter by specific documentation versions, like “OpenSearch 2.17 (latest)” or “OpenSearch 1.3.” This is particularly useful if you're working with different versions of OpenSearch and need access to historical documentation.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:56]:
```suggestion
    You can also filter by specific documentation versions, like "OpenSearch 2.17 (latest)" or "OpenSearch 1.3." This is particularly useful if you're working with different versions of OpenSearch and need access to historical documentation.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+
+- **Organized results display**:
+    Search results are displayed clearly, with:
+    - **Title and summary**:
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:60]: "A title and summary" (to complete the introductory stem)? Again, if this is an actual name, then fine to leave as is.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 8]
```diff
@@ -0,0 +1,71 @@
...
+    Each result includes a clickable title and a brief summary to give you an idea of the content before you click through.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:61]:
```suggestion
    Each result includes a clickable title and a brief summary of the content.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 9]
```diff
@@ -0,0 +1,71 @@
...
+    - **Breadcrumb navigation**:
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:62]: Above: "clickable" => "selectable"? (Accessibility)

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 10]
```diff
@@ -0,0 +1,71 @@
...
+    Each result shows its location within the documentation structure, helping you keep track of where you are.
```
Comment (natebower) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:63]: This sentence could probably end at "structure," but if we want to keep the second phrase, then "where you are" could perhaps be a bit more precisely stated.

**Review by natebower — COMMENTED** [review]
@kolchfa-aws @zelinh Editorial review complete. Nicely written! Please see my comments and changes and let me know if you have any questions.

Cc: @pajuric

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 7 above)
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:60]:
```suggestion
    - **A title and summary**:
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 10 above)
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:63]:
```suggestion
    Each result shows its location within the documentation structure.
```

**Comment by kolchfa-aws** [comment]
Thank you, @natebower! I addressed your comments.

**Comment by natebower** [comment]
> Thank you, @natebower! I addressed your comments.

Thanks @kolchfa-aws. LGTM!

Cc: @pajuric

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+We've also upgraded from basic query strings to `match` and `multi_match` queries, which provide more precise results by searching across multiple fields. Additionally, we've applied `indices_boost` to prioritize documentation in search results, further improving relevance.
+
+## New search results page
+
+We've moved away from the simple dropdown search interface in favor of a full search results page, offering a more powerful and user-friendly way to navigate search results. This page gives you more control over your search experience and makes it easier to refine your results.
+
+## Key features
+
+Here's a look at the key improvements and new features that enhance your search experience on OpenSearch.org:
```
Comment (smacrakis) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:41]: I think "that enhance your search experience" is telling the user what their reaction should be. I'd prefer to be more matter of fact, e.g., 

...new features in OpenSearch.org search:

**Comment by kolchfa-aws** [comment]
@pajuric Could you review/update the meta for this blog please?

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 12]
```diff
@@ -0,0 +1,71 @@
...
+
+- **Dedicated search results page**:
+    After submitting a search, you'll be taken to a new page that shows a more comprehensive list of results than was provided by the previous dropdown. You can browse through these results more easily, without leaving the page.
```
Comment (smacrakis) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:44]: Suggested text:
Search results are shown on a dedicated page with a comprehensive list of results, which makes it easy to review them.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 11 above)
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:41]:
```suggestion
Here's a look at the key improvements and new features in OpenSearch.org search:
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 12 above)
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:44]:
```suggestion
    Search results are shown on a dedicated page with a comprehensive list of results, which makes it easy to review them.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 13]
```diff
@@ -0,0 +1,71 @@
...
+   - zelinhao
+   - kolchfa
+date: 2024-10-14 
+categories:
+  - community-updates
+  - update
+  - community
+meta_keywords: OpenSearch search update, improved search results, advanced filters, documentation search, OpenSearch search page, Amazon OpenSearch Service, version-specific search, search infrastructure update
+meta_description: Discover the new and improved search functionality on OpenSearch.org. Learn how our updated infrastructure, advanced filters, and dedicated search results page make it easier to find the documentation, blog posts, and event information you need.
+---
```
Comment (pajuric) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:14]: Updated meta: 
meta_keywords: improve search functionality, OpenSearch search experience, OpenSearch.org, advanced filtering options, enhanced search experience

meta_description: Explore the enhanced search experience on OpenSearch.org. Discover how our upgraded infrastructure, advanced filtering options, and dedicated search page make it easier to locate the documentation and content you need.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 14]
```diff
@@ -0,0 +1,71 @@
...
+layout: post
+title: Introducing enhanced search functionality on OpenSearch.org
+authors:
+   - zelinhao
+   - kolchfa
+date: 2024-10-14
```
Comment (pajuric) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:7]: Update the date to: 2024-10-30

Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:7]:
```suggestion
date: 2024-10-30 
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 15]
```diff
@@ -0,0 +1,71 @@
...
+title: Introducing enhanced search functionality on OpenSearch.org
+authors:
+   - zelinhao
+   - kolchfa
+date: 2024-10-30 
+categories:
+  - community-updates
+  - update
+  - community
+meta_keywords: OpenSearch search update, improved search results, advanced filters, documentation search, OpenSearch search page, Amazon OpenSearch Service, version-specific search, search infrastructure update
```
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:12]:
```suggestion
meta_keywords: improve search functionality, OpenSearch search experience, OpenSearch.org, advanced filtering options, enhanced search experience
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown [hunk 16]
```diff
@@ -0,0 +1,71 @@
...
+meta_description: Discover the new and improved search functionality on OpenSearch.org. Learn how our updated infrastructure, advanced filters, and dedicated search results page make it easier to find the documentation, blog posts, and event information you need.
```
Comment (kolchfa-aws) [inline _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:13]:
```suggestion
meta_description: Explore the enhanced search experience on OpenSearch.org. Discover how our upgraded infrastructure, advanced filtering options, and dedicated search page make it easier to locate the documentation and content you need.
```

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 13 above)
Comment (kolchfa-aws) [reply _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown:14]: Updated.

File: _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown (hunk 14 above)
Comment (kolchfa-aws) [reply _posts/2024-10-14-Introduction-of-new-opensearch-website-search.markdown]: Done

**Comment by pajuric** [comment]
@krisfreedain - Blog is ready to publish. Is there a second maintainer I can loop in here to help push this?

**Review by krisfreedain — APPROVED** [review]
looks good!

**Pull request merged**